manager/assets/modext/widgets/core/modx.grid.js

```diff
@@ -495,14 +515,190 @@ Ext.extend(MODx.grid.Grid,Ext.grid.EditorGridPanel,{
+            permissions.push(modePermission);
+        });
+        this.showActionsMenu = permissions.length === 0 || permissions.every(permission => permission === false) === true
+            ? false
+            : true
+            ;
+    }
+
+    /**
+     * @property {Function} recordIsProtected - Used to remove the ability to delete
+     * specific record rows, regardless of permissions levels, based on a given record identifier
+     *
+     * @param {Number} subject - The value of the current record's identifier
+     * @param {Number} protectedIdentifiers - The record identifiers to be protected (making them non-editable/deletable)
+     *
+     * @return {Boolean}
+     */
+    ,recordIsProtected: function(subject, protectedIdentifiers) {
+        if (Ext.isEmpty(protectedIdentifiers)) {
+            return false;
+        }
+        protectedIdentifiers = protectedIdentifiers.map(identifier => {
+            return typeof identifier === 'string' ? identifier.trim() : identifier ;
+        });
+        return protectedIdentifiers.includes(subject);
+    }
+
+    /**
+     * @property {Function} valueIsReserved - Wraps a grid value with a real or simulated link — a trigger that appears
+     * like an anchor link, usually to access a dropdown chooser or other control
+     *
+     * @param {Array|String} reservedValues - A set of values that can not be used for a particular object's field
+     * @param {Object} value - The submitted value being tested
+     *
+     * @return {Boolean}
+     */
+    ,valueIsReserved: function(reservedValues, value) {
+        if (!Ext.isArray(reservedValues)) {
+            reservedValues = reservedValues.split(',');
+        }
+        let isReserved = false;
+        reservedValues.forEach(reserved => {
+            if (reserved.toLowerCase() === value.toLowerCase()) {
+                isReserved = true;
+                return;
+            }
+        });
+        return isReserved;
+    }
+
+    /**
+     * @property {Function} getRemovableItemsFromSelection - Prunes protected items from the current
+     * selection list before submitting for deletion, or for setting the state of the 'Delete Selected'
+     * menu item
+     *
+     * @param {String} itemIdType - The data type of the value being inspected (either string or integer)
+     *
+     * @return {Boolean|Array}
+     */
+    ,getRemovableItemsFromSelection: function(itemIdType = 'string') {
+        const   selectionList = this.getSelectedAsList(),
+                removableItems = []
+        ;
+        if (selectionList === false) {
+            return false;
+        }
+        selectionList.split(',').forEach(id => {
+            id = itemIdType === 'string' ? id : parseInt(id) ;
+            if (!this.recordIsProtected(id, this.protectedIdentifiers) && !this.nonRemoveableRecords.includes(id)) {
+                removableItems.push(id);
+            }
+        });
+        return removableItems.length > 0 ? removableItems : false ;
```

Comment:
```suggestion
       if (removableItems.length > 0) {
           return removableItems
       };
```

Its okay to return undefined. 
It not semantically correct to return boolean value.

Reply:
OK, I see what you're saying ... changed it a bit here and where called.